src/Serilog.Sinks.AzureDataExplorer/Sinks/AzureDataExplorerSink.cs

```diff
@@ -56,52 +54,76 @@ public AzureDataExplorerSink(AzureDataExplorerSinkOptions options)
+                m_ingestionMapping.IngestionMappingReference = mappingName;
             }
             else if (options.ColumnsMapping?.Any() == true)
             {
                 m_ingestionMapping.IngestionMappings = options.ColumnsMapping.Select(m => new ColumnMapping { ColumnName = m.ColumnName, ColumnType = m.ColumnType, Properties = new Dictionary<string, string>(1) { { MappingConsts.Path, m.ValuePath } } }).ToList();
             }
             else
             {
-                m_ingestionMapping.IngestionMappings = s_defaultIngestionColumnMapping;
+                m_ingestionMapping.IngestionMappings = SDefaultIngestionColumnMapping;
             }
 
             var kcsb = options.GetKustoConnectionStringBuilder();
+            var engineKcsb = options.GetKustoEngineConnectionStringBuilder();
 
             if (options.UseStreamingIngestion)
             {
-                m_queuedIngestClient = KustoIngestFactory.CreateStreamingIngestClient(kcsb);
+                m_ingestClient = KustoIngestFactory.CreateManagedStreamingIngestClient(engineKcsb, kcsb);
             }
             else
             {
-                m_queuedIngestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
+                m_ingestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
             }
         }
 
         public async Task EmitBatchAsync(IEnumerable<LogEvent> batch)
         {
             using (var dataStream = CreateStreamFromLogEvents(batch))
             {
-                var result = await m_queuedIngestClient.IngestFromStreamAsync(
-                    dataStream,
-                    new KustoIngestionProperties
-                    {
-                        DatabaseName = m_databaseName,
-                        TableName = m_tableName,
-                        Format = DataSourceFormat.multijson,
-                        IngestionMapping = m_ingestionMapping
-                    },
-                    new StreamSourceOptions
-                    {
-                        LeaveOpen = false,
-                        CompressionType = DataSourceCompressionType.GZip
-                    }).ConfigureAwait(false);
+                if (!m_streamingIngestion)
+                {
+                    await m_ingestClient.IngestFromStreamAsync(
+                        dataStream,
+                        new KustoQueuedIngestionProperties(m_databaseName, m_tableName)
+                        {
+                            DatabaseName = m_databaseName,
+                            TableName = m_tableName,
+                            FlushImmediately = m_flushImmediately,
+                            Format = DataSourceFormat.multijson,
+                            IngestionMapping = m_ingestionMapping
+                        },
+                        new StreamSourceOptions
+                        {
+                            LeaveOpen = false,
```

Comment:
Its better to add SourceId , this is useful in debugging. Same below as well.

We had the question that should we read the file as stream and flush it ? or we can use the API that is already present to flush it from File ? have we explored it ?

```
            var streamSourceOptions = new StreamSourceOptions()
            {
                SourceId = <A unique GUID>,
            };
```

Reply:
makes sense, helps us to debug in the DM side if required later. Added sourceId.